plone/scale/scale.py

```diff
@@ -129,31 +129,33 @@ def _scale_thumbnail(image, width=None, height=None):
     return image
 
 
-def scalePILImage(image, width=None, height=None, direction='down'):
+def scalePILImage(image, width=None, height=None, mode='contain', direction=None):
     """Scale a PIL image to another size.
 
     This is all about scaling for the display in a web browser.
 
     Either width or height - or both - must be given.
 
-    Three different scaling options are supported via `direction`:
+    Three different scaling options are supported via `mode` and correspond to
+    the CSS background-size values
+    (see https://developer.mozilla.org/en-US/docs/Web/CSS/background-size):
 
-    `up`
-        scaling scales the smallest dimension up to the required size
-        and crops the other dimension if needed.
-
-    `down`
+    `contain`
+        Alternative spellings: `scale-crop-to-fit`, `down`.
         scaling starts by scaling the largest dimension to the required
         size and crops the other dimension if needed.
 
-    `thumbnail`
-        scales to the requested dimensions without cropping. Theresulting
+    `cover`
+        Alternative spellings: `scale-crop-to-fill`, `up`.
+        scaling scales the smallest dimension up to the required size
+        and crops the other dimension if needed.
```

Comment:
For `up` my conclusion was:

"scaling scales the *largest* dimension up to the required size and crops the other dimension if needed."

Reply:
@mauritsvanrees I think you might be right. Could you take a look at #37 where I put the size calculations into a separate function, it might help to think this through better. With ``up`` and ``down`` you never get letter boxing but if the destination size doesn't fit it crops, the destinations sizes are never exceeded. With ``scale`` you will never get cropping, but the image might not fill the target dimensions and thus you get letter boxing.

Reply:
Should be fixed by: https://github.com/plone/plone.scale/pull/39/commits/e77aaa81e2be3f98d25e0c0bd2a33ff4ab086b9d